### PR TITLE
fix(BitField): use only enum names in iterating

### PR DESCRIPTION
**Pull request by jaw0r3k**
**Please describe the changes this PR makes and why it should be merged:**
Resolves: #9350
The change may seem weird, but its working.
As enums are reverse mapped, getting keys will not only give names but values
The shown exception was permissions, but its just an object

**Status and versioning classification:**
- Typings don't need updating
- This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: packages/discord.js/src/util/BitField.js [hunk 1]
```diff
@@ -140,7 +140,7 @@ class BitField {
 
   *[Symbol.iterator](...hasParams) {
     for (const bitName of Object.keys(this.constructor.Flags)) {
-      if (this.has(bitName, ...hasParams)) yield bitName;
+      if (typeof bitName === 'string' && this.has(bitName, ...hasParams)) yield bitName;
```
Comment (vladfrangu) [inline packages/discord.js/src/util/BitField.js:143]: I don't see how this fixes anything, considering the keys will always be strings 
Code - Insiders - 2023-04-09 at 18 57 24@2x

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: packages/discord.js/src/util/BitField.js (hunk 1 above)
Comment (jaw0r3k) [reply packages/discord.js/src/util/BitField.js]: Oh sorry if forgot to change the commit 
Now its working fine 
image

**Review by vladfrangu — APPROVED** [review]
sure

**Review by kyranet — APPROVED** [review]

**Comment by Jiralite** [comment]
It looks like this needs backporting.

**Comment by jaw0r3k** [comment]
> It looks like this needs backporting, would you mind doing it?

Sure <3

**Review by SpaceEEC — APPROVED** [review]

**Review by SpaceEEC — CHANGES_REQUESTED** [review]
On a second look, this needs to be applied to other places traversing `Flags` too.
This only seems to be `serialize` (currently).

---

```js
> new ActivityFlagsBitField(511).serialize()
{
  '1': true,
  '2': true,
  '4': true,
  '8': true,
  '16': true,
  '32': true,
  '64': true,
  '128': true,
  '256': true,
  Instance: true,
  Join: true,
  Spectate: true,
  JoinRequest: true,
  Sync: true,
  Play: true,
  PartyPrivacyFriends: true,
  PartyPrivacyVoiceChannel: true,
  Embedded: true
}
```

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**